test/lib/CustomRulePathManager.test.ts

```diff
@@ -0,0 +1,176 @@
+import { CustomRulePathManager, ENGINE } from '../../src/lib/CustomRulePathManager';
+import { FileIOHandler } from '../../src/lib/FileIOHandler';
+import { expect } from 'chai';
+
+import Sinon = require('sinon');
+// import { Messages } from '@salesforce/core';
+
+// Messages.importMessagesDirectory(__dirname);
+// const messages = Messages.loadMessages('scanner', 'add');
+
+// test cases:
+// read entries on initialization
+// don't run initialization if it was already run
+
+describe('CustomRulePathManager tests', () => {
+
+    let sandbox: Sinon.SinonSandbox;
+
+    before(() => { sandbox = Sinon.createSandbox(); });
+    afterEach(() => { sandbox.restore(); });
+
+    // File exists, but content has been somehow cleared
+    const emptyFile = '';
+
+    // One entry for apex, two for java
+    const populatedFile = '{"pmd": {"apex": ["/some/user/path/customRule.jar"],"java": ["/abc/def/ghi","/home/lib/jars"]}}';
+
+    describe('Rule path entries creation', () => {
+        it('should read CustomPaths.json to get Rule Path Entries', async () => {
+            // Setup stub
+            let stub = Sinon.createStubInstance(FileIOHandler);
+            stub.readFile.resolves(populatedFile);
+            const manager = new CustomRulePathManager(stub);
+
+            // Execute test
+            const rulePathMap = await manager.getRulePathEntries(ENGINE.PMD);
+            rulePathMap.forEach((value, key) => {
+                console.log(`${key}: ${Array.from(value)}`);
+            });
+
+            // Validate run
+            expect(stub.readFile.calledOnce).to.be.true;
+            expect(rulePathMap).to.be.lengthOf(2);
+
+            //Validate each entry
+            expect(rulePathMap).has.keys('apex', 'java');
+            const apexPaths = rulePathMap.get('apex');
+            expect(apexPaths).to.be.lengthOf(1);
+            expect(apexPaths).deep.contains('/some/user/path/customRule.jar');
+
+            const javaPaths = rulePathMap.get('java');
+            expect(javaPaths).to.be.lengthOf(2);
+            expect(javaPaths).deep.contains('/abc/def/ghi');
+            expect(javaPaths).deep.contains('/home/lib/jars');
+        });
+
+        it('should handle empty Rule Path file gracefully', async () => {
+            // Setup stub
+            let stub = Sinon.createStubInstance(FileIOHandler);
+            stub.readFile.resolves(emptyFile);
+            const manager = new CustomRulePathManager(stub);
+
+            // Execute test
+            const rulePathMap = await manager.getRulePathEntries(ENGINE.PMD);
+
+            // Validate
+            expect(rulePathMap).is.not.null;
+            expect(rulePathMap).to.be.lengthOf(0);
+
+        });
+
+        // it('should handle non-existent CustomPath.json file gracefully', async () => {
```

Comment:
Leaving this commented code that I'll clean up in the next pass. I'm still figuring out stub-throws-exception in Sinon.